include/memory/memory.hh

```diff
@@ -2,12 +2,41 @@
 #define __INCLUDE_MEMORY_MEMORY_HH__
 
 #include "common/common.hh"
+#include <iostream>
+#include <unordered_map>
 
 namespace sim {
 
 class Memory final {
 public:
-  Word load(std::size_t);
+  struct MemoryStats {
+
+    std::size_t numLoads;
+    std::size_t numStores;
+    std::size_t numPageFaults;
+
+    MemoryStats() : numLoads(0), numStores(0), numPageFaults(0) {}
+  };
+
+private:
+  std::unordered_map<Addr, Word> mem;
+  MemoryStats stats;
+
+  Word pageFaultHandle(Addr addr);
+
+public:
+  Memory();
```

Comment:
nit:
```suggestion
  Memory() = default;
```

Reply:
Added in ([2d0e439](https://github.com/106-inc/sim2022/pull/15/commits/2d0e439e617cd8b16865f4aaf87cdf1305cb98db))